tests/integration/remove_by_phase/test_remove_by_phase.py

```diff
@@ -60,3 +63,78 @@ def test_remove_flows_by_phase(mock_ask):
     paused_flows = asyncio.run(get_paused_flows())
 
     assert any([flow_id in paused_flows for flow_id in added_flows]) == False
+
+
+@patch('rich.prompt.Confirm.ask', return_value=True)
+def test_remove_flows_by_phase_and_list_with_empty_intersection(mock_ask):
+    initial_owned_flows = asyncio.run(get_serving_flows())
+    logger.info(f'Initial owned flows: {len(initial_owned_flows)}')
+    added_flows = []
+
+    for _ in range(NUM_FLOWS_TO_DEPLOY):
+        added_flows.append(
+            asyncio.run(
+                _simplified_deploy(CloudFlow(path=os.path.join(flows_dir, flow)))
+            )
+        )
+    logger.info(f'Deploying {NUM_FLOWS_TO_DEPLOY} new flows...')
+
+    owned_flows_after_add = asyncio.run(get_serving_flows())
+
+    logger.info(f'New Flow added: {added_flows}')
+    logger.info(f'Owned flows after new deployments: {len(owned_flows_after_add)}')
+    assert set(added_flows).issubset(owned_flows_after_add)
+
+    for i in range(NUM_FLOWS_TO_PAUSE):
+        asyncio.run(CloudFlow(flow_id=added_flows[i]).pause())
+
+    logger.info(f'Removing new flows by phase and list...')
+
+    args = Namespace()
+    args.phase = 'Paused'
+    args.flows = added_flows[NUM_FLOWS_TO_PAUSE:]
+    remove(args)
+
+    paused_flows = asyncio.run(get_paused_flows())
+    owned_flows_after_add = asyncio.run(get_serving_flows())
+
+    assert any([flow_id in owned_flows_after_add for flow_id in added_flows]) == False
+    assert any([flow_id in paused_flows for flow_id in added_flows]) == False
+
+
+@patch('rich.prompt.Confirm.ask', return_value=True)
+def test_remove_flows_by_phase_and_list_with_non_empty_intersection(mock_ask):
+    initial_owned_flows = asyncio.run(get_serving_flows())
+    logger.info(f'Initial owned flows: {len(initial_owned_flows)}')
+    added_flows = []
+
+    for _ in range(NUM_FLOWS_TO_DEPLOY):
+        added_flows.append(
+            asyncio.run(
+                _simplified_deploy(CloudFlow(path=os.path.join(flows_dir, flow)))
+            )
+        )
+
+    logger.info(f'Deploying {NUM_FLOWS_TO_DEPLOY} new flows...')
+    owned_flows_after_add = asyncio.run(get_serving_flows())
+
+    logger.info(f'New Flow added: {added_flows}')
+    logger.info(f'Owned flows after new deployments: {len(owned_flows_after_add)}')
+
+    assert set(added_flows).issubset(owned_flows_after_add)
+
+    for i in range(NUM_FLOWS_TO_PAUSE):
+        asyncio.run(CloudFlow(flow_id=added_flows[i]).pause())
+
+    logger.info(f'Removing new flows by phase and list...')
+
+    args = Namespace()
+    args.phase = 'Paused'
+    args.flows = added_flows[NUM_FLOWS_TO_PAUSE - 1 :]
+    remove(args)
+
+    paused_flows = asyncio.run(get_paused_flows())
+    owned_flows_after_add = asyncio.run(get_serving_flows())
+
+    assert any([flow_id in owned_flows_after_add for flow_id in added_flows]) == False
```

Comment:
can you add some comments what's being done here on high level

Reply:
@zac-li I have added some comments, let me know if they resolve this